src/XPToken.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.26;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { IXPProvider } from "./interfaces/IXPProvider.sol";
+
+contract XPToken is Ownable {
+    string public constant name = "XP Token";
+    string public constant symbol = "XP";
+    uint256 public constant decimals = 18;
+
+    uint256 public totalSupply;
+
+    IXPProvider[] public xpProviders;
+
+    error XPToken__TransfersNotAllowed();
+    error XPProvider__IndexOutOfBounds();
+
+    constructor(uint256 _totalSupply) Ownable(msg.sender) {
+        totalSupply = _totalSupply;
+    }
+
+    function setTotalSupply(uint256 _totalSupply) external onlyOwner {
+        totalSupply = _totalSupply;
+    }
```

Comment:
Wonder if this makes sense or if we should just go with a max uint total supply...

Reply:
as we decide we are going to merge this and create a new ticket to decide the new way to distribute the amounts